deepspeed/runtime/activation_checkpointing/checkpointing.py

```diff
@@ -402,6 +402,11 @@ def forward(ctx, run_function, *args):
                         contiguous_data_buffers[i] = tensor_list
                         data_offsets[i] = 0
 
+                    contiguous_data_buffers[i][data_offsets[i]].data[range(

```

Comment:
Why do we need this for the overlap_comm ? Could you explain what is hppening here?

Reply:
Actually this is irrelevant for the overlap_comm. This is for the second optimization mentioned in the post basically 'touching' the pages before the cudaMemcpy.

Reply:
Got it. This is a clever way to page in the memory. I think its fine to pin this memory directly though which might be cleaner?

Reply:
I think pinning the memory makes more sense, but I wasn't sure if there is any reason not to pin in the first place. And to take the full advantage of the pinned memory and some asynchronous behavior, there should be some more structural changes.

Reply:
Sure. The current solution is fine too. Could you please just add a comment there though on why that line exists?

Reply:
I added some comments for this step.